backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java

```diff
@@ -60,12 +60,20 @@
           .append(ModelDBConstants.DATASET_ID)
           .append(" = :datasetId AND dsv.")
           .append(ModelDBConstants.VERSION)
-          .append(" =:version")
+          .append(" =:version ")
+          .append(" AND dsv.")
+          .append(ModelDBConstants.DELETED)
+          .append(" = false ")
+          .toString();
+  private static final String CHECK_DATASET_VERSION_EXISTS_BY_ID_HQL =
+      new StringBuilder("Select count(*) From DatasetVersionEntity dsv where ")
+          .append(" dsv." + ModelDBConstants.ID + " = :datasetVersionId ")
+          .append(" AND dsv." + ModelDBConstants.DELETED + " = false ")
           .toString();
   private static final String DATASET_VERSION_BY_DATA_SET_IDS_QUERY =
       "From DatasetVersionEntity ds where ds.dataset_id IN (:datasetIds) ";
   private static final String DATASET_VERSION_BY_IDS_QUERY =
-      "From DatasetVersionEntity ds where ds.id IN (:ids)";
+      "From DatasetVersionEntity ds where ds.id IN (:ids) AND ds.deleted = false ";
```

Comment:
Should we use ModelDBConstants.DELETED here too?

Reply:
updated with constant name